lib/logstash/devutils/rspec/logstash_helpers.rb

```diff
@@ -36,32 +37,45 @@ def sample(sample_event, &block)
         pipeline.instance_eval { @filters.each(&:register) }
 
         event.each do |e|
-          pipeline.filter(e) {|new_event| results << new_event }
+          pipeline.filter(e) {|new_event| results << new_event}
         end
 
         pipeline.flush_filters(:final => true) do |e|
           results << e unless e.cancelled?
         end
 
-        results
+        results.select{|e| !e.cancelled?}
```

Comment:
no, flush only takes care of  the flushed events if any. we could "optimize" and rewrite the flushing to be like
```ruby
         pipeline.flush_filters(:final => true) do |e|
           results << e
         end
```
but in the end it does not fix anything,